packages/grafana-ui/src/components/Link/TextLink.tsx

```diff
@@ -0,0 +1,106 @@
+import { css } from '@emotion/css';
+import React, { AnchorHTMLAttributes, forwardRef } from 'react';
+
+import { GrafanaTheme2, locationUtil, textUtil, ThemeTypographyVariantTypes } from '@grafana/data';
+
+import { useTheme2 } from '../../themes';
+import { IconName, IconSize } from '../../types';
+import { Icon } from '../Icon/Icon';
+import { customColor, customWeight } from '../Text/utils';
+
+import { Link } from './Link';
+
+interface TextLinkProps extends Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 'target' | 'rel'> {
+  /** url to which redirect the user, external or internal */
+  href: string;
+  /** Color to use for text */
+  color?: keyof GrafanaTheme2['colors']['text'];
+  /** Specify if the link will redirect users to a page in or out Grafana */
+  external?: boolean;
+  /** True if the element should be displayed inline with surrounding text, false if it should be displayed as a block */
+  inline?: boolean;
+  /** What typograpy variant should be used for the component. Only use if default variant for the defined element is not what is needed */
```

Comment:
I think we can remove "Only use if default variant for the defined element is not what is needed " as this only makes sense for `Text` component right?

Reply:
Do we need this inline option? Not sure i understand the need for it. I would assume this component would default to inline. And if I wanted something else I would just place it as a child of a flex container

Reply:
We apply one default style or another depending on the `inline` prop. It was not well explained in the description so I changed it.

Reply:
But why do we need it? Can't it be the default? Flex containers automatically make their children non inline (even if we they are).

I don't see the reason for this. Think this component can be inline-flex always. It should never be a plain block element

Reply:
Hi @torkelo,
I took a look at how the `Button` component is used, being always `inline-flex`, and I see your point. I will make the changes needed so it will be `inline-flex` always as well and we improve consistency.

Regarding why we need the `inline` prop, we set styles depending on if the link will be wrapped by more text or a standalone element as you can see in [this comment](https://github.com/grafana/grafana/pull/69330#discussion_r1238336842). 

We, @staton-hyse11 and the rest of the design system team, had a discussion and agreed on basing the differentiation on this prop, so:
1) If `inline = true` the _text link_ will be blue and underlined at first while the underline disappears on hover.
2) If `inline = false` the _text link_ will be 'primary' and not underlined while underlined on hover.
Through the color prop the user will be able to change the color value, but the rest will be kept. 
Example: 
```
<TextLink href='/' color='link'>Example</TextLink>
```
The _text link_ will be blue, as set by the user, and not underlined at first while blue and underlined when onhover, as it inherits from the standalone default behaviour.

Reply:
but we have inline links that are not underline in many places today (Page descriptions and other places). When to set inline=true? What makes a link inline? if inline controls color why have a color option? 

This seems like it will just cause confusion (when to set to true) and incompatible and outright really bad (accessibility-wise) links when inline prop & color prop is used in bad combos. 

hmm.. maybe if we rename it "underline" it would be a bit less confusing? not sure. maybe it's just me.  

But don't see this view as any blockers, if the design team wants it this way. Probably just me who finds inline & color prop confusing (since all links are by default inline). Should one blue link be underline while one next to it be not (if one is in a flex container and the other is not)?

Reply:
@torkelo from our documentation inline means: 
- is part of a broader set of text. 
- It will be surrounded by text, before, after, or in both directions.

Today's implementation of links within text can be improved on, that's what this change will provide. We are making the distinction between links that occur within bodies of text or by themselves like blog post titles.

Links in different implementations will need different affordances to ensure they're known to be links. Today we have varying colors, varying decorations for links in different locations. This will reduce that. I will ilnk our documentation in our Slack channel and further conversations around how we are standardizing this component should be had there.

Reply:
@staton-hyse11 ok, thanks for explaining. Makes some sense. 

Just got a bit confused with how color prop can override it, and part of me feels like having blue links with different styling is a bit too complex. 

I like how GitHub have
* All blue links are not underlined but have underlined on hover (no matter if it's by itself or inside text). 
* Primary text links (white/dark dep on theme) have blue + underline hover  (like this combo enforcing it's a link) 
* Secondary text links (muted) have blue + underline hover  

There are some exceptions perhaps but that seems to be the default variants.

To me having variants like the above look more simple and consistent than color + inline prop (an inline prop that seems to dictate the color and underline/hover behavior). It feels simpler / more consistent to me if the styling & hover effect of the link is defined by its default variant/color, not by its location on the page. 

But just a thought / perspective

Reply:
but let's continue with color + inline prop, don't want to derail anything (sorry, keep doing it...)